terragrunt/aws/glue/etl/platform/gc_forms/process_data.py

```diff
@@ -106,6 +114,42 @@ def is_type_compatible(series: pd.Series, glue_type: str) -> bool:
     return True
 
 
+def validate_with_gx(dataframe: pd.DataFrame, checkpoint_name: str) -> bool:
+    """
+    Validate the DataFrame using the specified Great Expectations checkpoint.
+    Logs detailed errors if validation fails.
+    """
+    gx_context_path = os.path.join(os.path.dirname(__file__), "gx")
+    context = gx.get_context(context_root_dir=gx_context_path)
```

Comment:
Can you test something here.  I'm trying to figure out why we're getting DNS queries in our workflow runs to `greatexpectations.io` and I suspect it's the library looking for it's GX Cloud config:
```suggestion
    context = gx.get_context(context_root_dir=gx_context_path, cloud_mode=False)
```